x-pack/plugins/lens/public/shared_components/legend_settings_popover.tsx

```diff
@@ -158,6 +212,63 @@ export const LegendSettingsPopover: React.FunctionComponent<LegendSettingsPopove
         position={position}
         onPositionChange={onPositionChange}
       />
+      <EuiFormRow
+        display="columnCompressedSwitch"
+        label={i18n.translate('xpack.lens.shared.truncateLegend', {
+          defaultMessage: 'Truncate legend text',
```

Comment:
As the user is already in the legend menu, repeating the word "legend" here feels unnecessary. For the sake of brevity and alignment, consider changing to the following:

```suggestion
          defaultMessage: 'Truncate text',
```